controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -95,6 +114,124 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			// Wait (for `timeout` duration) to receive an acknowledgement from the provider server.
+			_, err = providerClient.FulfillStorageClassClaim(ctx, storagecluster.Status.ExternalStorage.ConsumerID, r.storageClassClaim.Name, r.storageClassClaim.Spec.EncryptionMethod, storageClassClaimStorageType)
+			if err != nil {
+				return err
+			}
+
+			/*	TODO
+				Periodically poll the provider server for resources provisioned for that claim.
+					Synchronous RPC since we need to wait for ACK to continue further
+					rpc PollProviderServer (PollProviderRequest) returns (PollProviderServerResponse) {}
+				Create (or do nothing if already created) the resources received from the provider server on the consumer cluster.
+			*/
+
+			// storageClass with the same name as storageClassClaim is controlled by the storageClassClaim.
+			// storageClass must be garbage collected when the respective storageClassClaim is deleted.
+			return controllerutil.SetControllerReference(r.storageClassClaim, storageClass, r.Scheme)
+		})
+
+		// Update the StorageClassClaim status.
+		r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimReady)
+
+	} else {
+
+		// Update the StorageClassClaim status.
+		r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimPending)
+
+		// Wait until all PVs using the StorageClass under deletion are removed.
+		// Check for any PVs using the StorageClass.
+		// TODO See if there is a better way to do this.
+		pvList := &corev1.PersistentVolumeList{}
+		err := r.Client.List(ctx, pvList)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		for pvi := range pvList.Items {
+			pv := &pvList.Items[pvi]
+			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
+				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by PV %s", r.storageClassClaim.Name, pv.Name)
+			}
+		}
+
+		// Update the StorageClassClaim status.
+		r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimDeleting)
+
+		// Delete StorageClass.
+		if err = r.get(ctx, storageClass); err == nil {
+			if err = r.delete(ctx, storageClass); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to delete StorageClass %s: %s", storageClass.Name, err)
+			}
+		} else if errors.IsNotFound(err) {
+			log.Info("StorageClass already deleted.")
+		} else {
+			return reconcile.Result{}, fmt.Errorf("failed to get StorageClass %s: %s", storageClass.Name, err)
+		}
+
+		// Call `RevokeStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+		/*  TODO
+		Wait (for `timeout` duration) to receive an acknowledgement from the provider server.
+		ACK response from the provider server tells client that provider will now delete the resources.
+		*/
+		_, err = providerClient.RevokeStorageClassClaim(ctx, storagecluster.Status.ExternalStorage.ConsumerID, r.storageClassClaim.Name)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
No timeout please, same reasoning as with the fulfill

Reply:
Added a TODO for both of the client fulfill and revoke methods. Need to have a discussion around this.